### PR TITLE
Add support for `SUBDATE()`

### PR DESCRIPTION
**Pull request by fulghum**
Implements [the `SUBDATE()` built-in function](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_subdate) and cleans up a few other spots. 

Related to: https://github.com/dolthub/dolt/issues/7277

### CODE REVIEW

File: enginetest/queries/queries.go [hunk 1]
```diff
@@ -4213,6 +4213,14 @@ Select * from (
...
 		Expected: nil,
 	},
+	{
+		Query:    "SELECT id FROM typestable WHERE da >= subdate('2020-01-01', INTERVAL 1 DAY)",
+		Expected: []sql.Row{{int64(1)}},
+	},
+	{
+		Query:    "SELECT id FROM typestable WHERE da >= subdate('2020-01-01', 1)",
+		Expected: []sql.Row{{int64(1)}},
+	},
```
Comment (jycor) [inline enginetest/queries/queries.go:4223]: Can you add enginetests that use columns from a table
Something like:
```sql
select subdate(i, j) from t;
select subdate(i, concat(j)) from t;
select subdate(i, j / 100) from t;
```

**Review by jycor — APPROVED** [review]
LGTM could use some enginetests

Here are some other differences (I don't think these are blocking, but are related enough to mention):

We display a datetime, when MySQL only displays a date (I don't think we distinguish between datetime, date, and time in general)

MySQL
```sql
mysql> select subdate('2008-01-02', 12);
+---------------------------+
| subdate('2008-01-02', 12) |
+---------------------------+
| 2007-12-21                |
+---------------------------+
1 row in set (0.0004 sec)
```

Dolt:
```sql
tmp/main*> select subdate('2008-01-02', 12);
+---------------------------+
| subdate('2008-01-02', 12) |
+---------------------------+
| 2007-12-21 00:00:00       |
+---------------------------+
1 row in set (0.00 sec)

tmp/main*> select date_sub('2008-01-02', interval 12 day);
+-----------------------------------------+
| date_sub('2008-01-02', interval 12 day) |
+-----------------------------------------+
| 2007-12-21 00:00:00                     |
+-----------------------------------------+
1 row in set (0.00 sec)
```

We fully evaluate this, when mysql truncates
MySQL
```sql
mysql> select subdate('2008-01-02', '12e10');
+--------------------------------+
| subdate('2008-01-02', '12e10') |
+--------------------------------+
| 2007-12-21                     |
+--------------------------------+
1 row in set, 1 warning (0.0004 sec)
Warning (code 1292): Truncated incorrect INTEGER value: '12e10'
```

Dolt:
```sql
tmp/main*> select subdate('2008-01-02', '12e10');
+--------------------------------+
| subdate('2008-01-02', '12e10') |
+--------------------------------+
| 1770-07-19 03:54:13.485772     |
+--------------------------------+
1 row in set (0.00 sec)
```

Apparently datetimes can overflow
MySQL
```sql
mysql> select subdate('2008-01-02', 12e10);
+------------------------------+
| subdate('2008-01-02', 12e10) |
+------------------------------+
| NULL                         |
+------------------------------+
1 row in set, 1 warning (0.0004 sec)
Warning (code 1441): Datetime function: datetime field overflow
```

Dolt:
```sql
tmp/main*> select subdate('2008-01-02', 12e10);
+------------------------------+
| subdate('2008-01-02', 12e10) |
+------------------------------+
| 1770-07-19 03:54:13.485772   |
+------------------------------+
1 row in set (0.00 sec)
```

File: enginetest/queries/queries.go (hunk 1 above)
Comment (fulghum) [reply enginetest/queries/queries.go:4223]: Good suggestion to use more complex expressions. Added in a6714fe79dc392d65e9160048b857f21e663db9a

**Comment by fulghum** [comment]
Thanks for the thorough review! You found some good behavior differences there between Dolt and MySQL. I dug into these a little more and opened tracking issues for them:
- https://github.com/dolthub/dolt/issues/7304
- https://github.com/dolthub/dolt/issues/7302
- https://github.com/dolthub/dolt/issues/7300

**Pull request merged**